### ISSUE TITLE
Graph analysis - Maddi

### ISSUE THREAD

**Issue by madshowie** [itchyshin/bibliometrics_2022#14]
Visualisation to do list:

- [x] Basic graphs from IUCN - species entries, stacked bar plot of threatened %, stacked bar plot of populationTrend %*
- [ ] Publication trend graph (from Scopus)
- [ ] Want to compare geographical distribution with geographical productivity of publications - map of species distribution can be made from the IUCN data but unsure how to incorporate productivity to show both in one graph. Might need to show two and compare side by side. 
- [ ] Order focus in publications vs. described species - unsure how to merge the two**
- [ ] Order focus in publications vs threatened species - unsure how to merge the two**
- [ ] Order focus along timeline - would like to do a term map that is colour coded with a year gradient (if possible)

*This is some of the basic plots I've drafted, hopefully they are useful. I'm just unsure how I can merge this information with the Scopus data (they're not complete with titles just yet, I just thought I'd pop them up here).

![Screen Shot 2022-07-11 at 6 48 20 pm](https://user-images.githubusercontent.com/106729043/178225467-09860589-41ea-4ade-a212-e02d17201d3b.png)
![Screen Shot 2022-07-11 at 6 47 58 pm](https://user-images.githubusercontent.com/106729043/178225475-ea87e627-2508-41f0-adca-fff4d696a913.png)
![Screen Shot 2022-07-11 at 6 47 47 pm](https://user-images.githubusercontent.com/106729043/178225477-713ed444-c6c6-46f6-8ead-adc93d1f899e.png)

**I would like to combine publications and described/threatened species in something like below. I'm unsure if this will work and if there's a better way to show this. 

![Screen Shot 2022-07-11 at 6 33 59 pm](https://user-images.githubusercontent.com/106729043/178224295-56847819-a2f6-47a6-9bde-8bd4b1302a4c.png)

**Comment by mlagisz** [comment]
You can see the code for merging data written by Jess here https://github.com/jessicatytam/biases_in_mammalian_research/blob/main/R/add_categories_for%20model.R
She does merging on line 72, but this is preceded by quite a bit of data wrangling, so she can match species names and also some prep for coding the various categories of data she has in both data frames.

**Comment by madshowie** [comment]
Thanks Losia. I'm still a bit confused with this though, do you know why she added the h-index? Also, all my labels are in the one column, and I'm not too sure how I can separate these to look at them individually. Do you have any suggestions?

**Comment by madshowie** [comment]
Or perhaps I can just leave them as is, if there's a way to select individual parts of the cell (but that may be difficult).

**Comment by mlagisz** [comment]
Jess's project was about calculating and analysing distribution of h-index
across mammalian species.

To separate strings at specific delimiter (e.g. "," or ", ") you can use
function strsplit (
https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/strsplit),
but you may find str_split and str_spl_fixed from the stringr package (
https://stringr.tidyverse.org/reference/str_split.html) more useful.

**Comment by madshowie** [comment]
I've managed to split the labels however it has only developed a list. Do you know how I could add these as separate columns to the original data frame?

**Comment by mlagisz** [comment]
you can use str_split_fixed: Split up a string into a fixed number of
pieces.

**Comment by madshowie** [comment]
Ohh yes ok, thank you that helped! Now I think I've made a mistake with my screening :( 

It seems some of the labels are in the wrong columns - do you know if there's a way to fix this in R (swap labels between columns) or if it best to quickly change these over in Rayyan and re-import?

Example - 'mining' should be in the second column and 'Amphibia' should be in the first. 
![Screen Shot 2022-07-13 at 1 51 02 pm](https://user-images.githubusercontent.com/106729043/178646898-4cf4777c-5ff4-4632-8e8e-bd084851b50c.png)

**Comment by mlagisz** [comment]
If it is just a few mistakes, you can change them in R, e.g. before
splitting the labels you can substitutes some using replace (dplyr,
https://statisticsglobe.com/r-replace-value-of-data-frame-variable-dplyr-package
or "gsub" (base, https://www.programmingr.com/tutorial/gsub-in-r/)
functions.

**Comment by madshowie** [comment]
Ok thank you Losia. After using str_split_fixed, the labels have been split but stored into a character matrix so I can't seem to use this data for graphs. Is there a way to place this into the original data.frame?

**Comment by mlagisz** [comment]
If the matrix is in the same order, you can just bind it as new columns to
the original data table (https://dplyr.tidyverse.org/reference/bind.html)
Also see
https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf

**Comment by madshowie** [comment]
Hi Losia, thanks for that I was able to add the species labels to the Scopus data frame. However I am still struggling to merge the IUCN df with the Scopus df. I'm finding it hard to understand how I can merge the two when they don't really have a common variable i.e. I can't 'join by (variable)' because they don't have it in common. I tried to use Jess's code but I'm a bit unsure what she's done without seeing her data frames. Do you have any suggestions? Is it best to just keep them separate?

**Comment by mlagisz** [comment]
Hi, you can merge dataframes when each of them has a column with
(partially) overlapping values - in this case it would be the species names
(also in Jess's project).
However the species names have to be in the same format e.g. having one
dataframe with "Rana temporaria" and the other with "Rana_temporaria"  will
not work. It is good to have a look at the
columns with species names in both data frames to see if they are formatted
the same or differently. Then do some re-formatting/cleaning as needed
(best save processed values as a new column in the data frame).

It is easiest to merge data frames if both have the same equivalent columns
with exactly identical column names, e.g. in both this column is named
"Species_Latin".
The "join" function family in tidyverse can detect such identically names
column, but it is safest to tell it explicitly which column to use by using
parameter "by", e.g. by = Species_Latin. This parameter can take two
column names, in case they are named differently e.g. "species" and
"Species".
Hope this helps.

**Comment by madshowie** [comment]
Hi Losia, that's great, thank you. I may have an issue though because the species labels in the Scopus data frame are in two separate columns (if there were more than two species I labelled as multiple species). Is it best to go through Rayyan and re-label any abstracts with more than 1 species as 'multiple species' so that I've only got one column?

![Screen Shot 2022-07-15 at 10 12 47 am](https://user-images.githubusercontent.com/106729043/179121580-36848b13-f67a-4c11-a707-0f2ca0fd629a.png)
